tests/migrations/test_loader.py

```diff
@@ -48,6 +54,28 @@ def test_apply(self):
             set(),
         )
 
+    def test_has_table_cached(self):
+        """
+        Test that the has_table() method caches a positive result and does not
+        continually query for the existence of the migrations table.
+        """
+        recorder = MigrationRecorder(connection)
+        # Ensure that the recorder table does not exist
+        with connection.schema_editor() as editor:
+            editor.delete_model(recorder.Migration)
+
+        with self.assertNumQueries(2):
+            self.assertEqual(recorder.has_table(), False)
+            self.assertEqual(recorder.has_table(), False)
+
+        # Create the table
+        recorder.ensure_schema()
+
+        # Only one query should be created
```

Comment:
Does this comment has any value? :thinking: as this is exactly what `assertNumQueries()` does.
 ```suggestion
        # Only one query should be created
```